driver/src/main/java/org/neo4j/driver/Session.java

```diff
@@ -87,67 +88,169 @@
      * @param work the {@link TransactionWork} to be applied to a new read transaction.
      * @param <T> the return type of the given unit of work.
      * @return a result as returned by the given unit of work.
+     * @deprecated superseded by {@link #executeRead(TransactionCallback)}.
      */
+    @Deprecated
     <T> T readTransaction( TransactionWork<T> work );
 
     /**
-     * Execute a unit of work in a managed {@link AccessMode#READ read} transaction
-     * with the specified {@link TransactionConfig configuration}.
+     * Execute a unit of work as a single managed transaction with {@link AccessMode#READ read} access mode and retry behaviour.
      * <p>
-     * This transaction will automatically be committed unless an exception is
-     * thrown during query execution or by the user code.
+     * The driver will attempt committing the transaction when the provided unit of work completes successfully. A user initiated failure of the unit of work
```

Comment:
```suggestion
     * The driver will attempt committing the transaction when the provided unit of work completes successfully. Any runtime exception thrown by the unit of work
```

Reply:
Thanks. I have made this part a bit more detailed.